src/bgw/scheduler.c

```diff
@@ -60,6 +60,16 @@ static bool jobs_list_needs_update;
 /* has to be global to shutdown jobs on exit */
 static List *scheduled_jobs = NIL;
 
+static MemoryContext scratch_mctx;
+
+/*
+ * macro to make sure we stay in scratch memory context because CommitTransactionCommand will
+ * reset memory context to TopMemoryContext
+ */
+#define CommitTransactionCommand()                                                                 \
```

Comment:
I strongly suggest to make the name more explicit about what it does, without overriding the original command or not use this macro at all